hibernate-core/src/main/java/org/hibernate/sql/results/internal/StandardRowReader.java

```diff
@@ -134,14 +138,22 @@ private void coordinateInitializers(RowProcessingState rowProcessingState) {
 			}
 		}
 
-		for ( int i = 0; i < numberOfInitializers; i++ ) {
+		final Initializer rootInitializer = initializers.get( numberOfInitializers - 1 );
```

Comment:
```suggestion
		final int lastIndex = numberOfInitializers - 1;
		final Initializer rootInitializer = initializers.get( lastIndex );
```

and use `lastIndex` in the loop conditions below